qiskit_nature/operators/second_quantization/quadratic_hamiltonian.py

```diff
@@ -0,0 +1,351 @@
+                be specified.
+            ValueError: Hermitian part and antisymmetric part must have same shape.
+            ValueError: Hermitian part must have shape num_modes x num_modes.
+            ValueError: Hermitian part must be Hermitian.
+            ValueError: Antisymmetric part must have shape num_modes x num_modes.
+            ValueError: Antisymmetric part must be antisymmetric.
+        """
+        if num_modes is not None:
+            self._num_modes = num_modes
+        elif hermitian_part is not None:
+            self._num_modes, _ = hermitian_part.shape
+        elif antisymmetric_part is not None:
+            self._num_modes, _ = antisymmetric_part.shape
+        else:
+            raise ValueError(
+                "Either Hermitian part, antisymmetric part, or number of modes must be specified."
+            )
+
+        if validate:
+            if (
+                hermitian_part is not None
+                and antisymmetric_part is not None
+                and hermitian_part.shape != antisymmetric_part.shape
+            ):
+                raise ValueError("Hermitian part and antisymmetric part must have same shape.")
+            if hermitian_part is not None:
+                if hermitian_part.shape[0] != self._num_modes:
+                    raise ValueError("Hermitian part must have shape num_modes x num_modes.")
+                if not _is_hermitian(hermitian_part, rtol=rtol, atol=atol):
+                    raise ValueError("Hermitian part must be Hermitian.")
+            if antisymmetric_part is not None:
+                if antisymmetric_part.shape[0] != self._num_modes:
+                    raise ValueError("Antisymmetric part must have shape num_modes x num_modes.")
+                if not _is_antisymmetric(antisymmetric_part, rtol=rtol, atol=atol):
+                    raise ValueError("Antisymmetric part must be antisymmetric.")
+
+        self._hermitian_part = hermitian_part
+        self._antisymmetric_part = antisymmetric_part
+        self._constant = constant
+
+        if self._hermitian_part is None:
+            self._hermitian_part = np.zeros((self._num_modes, self._num_modes))
+        if self._antisymmetric_part is None:
+            self._antisymmetric_part = np.zeros((self._num_modes, self._num_modes))
+
+    @property
+    def hermitian_part(self) -> np.ndarray:
+        """The matrix of coefficients of terms that conserve particle number."""
+        return self._hermitian_part
+
+    @property
+    def antisymmetric_part(self) -> np.ndarray:
+        """The matrix of coefficients of terms that do not conserve particle number."""
+        return self._antisymmetric_part
+
+    @property
+    def constant(self) -> float:
+        """The constant."""
+        return self._constant
+
+    @property
+    def num_modes(self) -> float:
+        """The number of modes this operator acts on."""
+        return self._num_modes
+
+    def to_fermionic_op(self) -> FermionicOp:
```

Comment:
I am wondering if we should rename this to `to_second_q_op` which is inline with the namings in the `properties` module.
If we (in the future) design a common base class for some matrix container that will likely be the name of this function, avoiding the deprecation of this one.

Reply:
I think the current name is much better because it precisely describes what the method does. `to_second_q_op` is confusing because QuadraticHamiltonian is already a "second-quantized op" (though not a SecondQuantizedOp). From the perspective of a user trying to discover this method, I think they are more likely to guess the current name.